doc/devel/release_guide.rst

```diff
@@ -1,169 +1,273 @@
+need to create a python2 environment with ``requests==2.9.0`` and linkchecker ::
+
+  conda create -p /tmp/lnkchk python=2 requests==2.9.0
+  source activate /tmp/lnkchk
+  pip install linkchecker
+  pushd doc/build/html
+  linkchecker index.html --check-extern
 
-* Run all of the regression tests by running ``python tests.py``
-  at the root of the source tree.
+Address any issues which may arise.  The internal links are checked on travis, this should only
+flag failed external links.
 
-* Run :file:`unit/memleak_hawaii3.py` and make sure there are no
-  memory leaks
+.. _release_tag:
 
-* try some GUI examples, e.g., :file:`simple_plot.py` with GTKAgg, TkAgg, etc...
+Create release commit and tag
+-----------------------------
 
-* remove font cache and tex cache from :file:`.matplotlib` and test
-  with and without cache on some example script
+To create the tag, first create an empty commit with a very terse set of the release notes
+in the commit message ::
 
-* Optionally, make sure :file:`examples/tests/backend_driver.py` runs
-  without errors and check the output of the PNG, PDF, PS and SVG
-  backends
+  git commit --allow-empty
 
-.. _release-branching:
+and then create a signed, annotated tag with the same text in the body
+message ::
 
-Branching
-=========
+  git tag -a -s v2.0.0
 
-Once all the tests are passing and you are ready to do a release, you
-need to create a release branch.  These only need to be created when
-the second part of the version number changes::
+which will prompt you for your gpg key password and an annotation.
+For pre releases it is important to follow :pep:`440` so that the
+build artifacts will sort correctly in pypi.  Finally, push the tag to GitHub ::
 
-   git checkout -b v1.1.x
-   git push git@github.com:matplotlib/matplotlib.git v1.1.x
+  git push -t DANGER v2.0.0
 
-On the branch, do any additional testing you want to do, and then build
-binaries and source distributions for testing as release candidates.
+Congratulations, the scariest part is done!
 
-For each release candidate as well as for the final release version,
-please `git tag` the commit you will use for packaging like so::
+To prevent issues with any down-stream builders which download the
+tarball from GitHub it is important to move all branches away from the commit
+with the tag [#]_::
 
-    git tag -a v1.1.0rc1
+  git commit --allow-empty
+  git push DANGER master
 
-The `-a` flag will allow you to write a message about the tag, and
-affiliate your name with it. A reasonable tag message would be something
-like ``v1.1.0 Release Candidate 1 (September 24, 2011)``. To tag a
-release after the fact, just track down the commit hash, and::
 
-    git tag -a v1.0.1rc1 a9f3f3a50745
+.. [#] The tarball that is provided by GitHub is produced using `git
+       archive <https://git-scm.com/docs/git-archive>`__.  We use
+       `versioneer <https://github.com/warner/python-versioneer>`__
```

Comment:
`versioneer` is a bit of a red-herring; it's just the `git archive` process that produces the tag/branch info. I just tried it and now I see what you mean about it.